### ISSUE TITLE
File names are set incorrectly 

### ISSUE THREAD

**Issue by sfwtn** [flathub/net.blockbench.Blockbench#50]
# Expected behavior
When saving files in Blockbench, file name saved on disk should be the same as previewed.

# Actual behavior
Saving any file adds `.xdp-` at the beginning and `-[6 char random string]` at the end of any file name.
For example:
`test.bbmodel` is saved as `.xdp-test.bbmodel-LHyoVq` on disk.
Because of `.xdp-` all saved files are hidden by default, which means they can't even be opened in Blockbench via the standard open menu.

# Steps to reproduce
1 - Create a new model
2 - Save it to disk
3 - Check the file name

This issue applies to any file type, including png textures, json models, etc.

# System info
Operating System: Arch Linux 
KDE Plasma Version: 5.27.10
Kernel Version: 6.6.13-1-lts (64-bit)
Graphics Platform: X11
Flatpak version: 1.15.6
Blockbench version: 4.9.3

**Comment by daenney** [comment]
All we do here is take the upstream Blockbench build and repackage it as a flatpak. I don't see how that could result in the described behaviour of:

> Saving any file adds .xdp- at the beginning and -[6 char random string] at the end of any file name.

If you can show that it works as expected with a non-flatpak build then we can take a look. But otherwise, this sounds like an upstream issue and not a packaged-in-flatpak issue.

**Comment by daenney** [comment]
There is a similar issue on JannisX11/blockbench#2186, suggesting we might be looking at some issue saving a temporary file and then renaming it to its final destination. That could be caused by a lack of permissions for where it's trying to save to. But I'm surprised some error doesn't show up.

**Comment by sfwtn** [comment]
> If you can show that it works as expected with a non-flatpak build then we can take a look. But otherwise, this sounds like an upstream issue and not a packaged-in-flatpak issue.

Sure. Here's a video of flatpak and appimage versions side by side

https://github.com/flathub/net.blockbench.Blockbench/assets/121581681/b9064e4d-dd3f-48be-ae1a-2c4f74ba6f07

> we might be looking at some issue saving a temporary file and then renaming it to its final destination. That could be caused by a lack of permissions for where it's trying to save to.

It's definitely temporal file name and my first thought was to give it higher filesystem access. However the result is the same even if I give it full system access.

**Comment by sfwtn** [comment]
Found the issue, it was on me. It needs `xdg-[directory]` permission, so to save in ~/Documents it needs `xdg-documents` permission. For ~/Downloads `xdg-downloads`, etc.
It is set by default, but seems I removed it at some point.
Another interesting discovery however is that if the path you're trying to save to is not covered by `xdg` it will run into the same temp file issue. For example I have a mounted drive in root and it still saves temp file name there. Might want to open an issue for that.
